collect_app/src/main/java/org/odk/collect/android/formentry/FormEntryViewModel.java

```diff
@@ -240,12 +238,12 @@ boolean isViewExiting() {
         private final SaveRequest saveRequest;
         private final FormSaver formSaver;
 
-        private final WeakReference<Listener> listenerRef;
+        private final Listener listener;
```

Comment:
Sounds good.

Reply:
Actually not sure we need these here especially. It would be nice to have them everywhere but also not something I see as practical (unless we had a lint for it). This is also a private class so isn't really a "public" API. Did you have a specific reason for the suggestion?

Reply:
Listeners can be optional but not in this case so it feels like a nice safeguard if someone needs to eventually be in this code. It seems like I've usually seen null checks around listener usage.

Also, when you had the `WeakReference` wrapper, there were clear null checks. So the diff that took that wrapper away made it very apparent that having a null listener would be a problem.

Don't feel strongly about it, though.